### ISSUE TITLE
Increase Airflow logging coverage, review existing logs and add missing essential logs

### ISSUE THREAD

**Issue by omkar-foss** [apache/airflow#43180]
### Description

As per users' feedback in the Airflow Debugging Survey 2024, about 42% of respondents mentioned that logs are missing or non-existent in some cases.

### Use case/motivation

Goal for this issue is to review the existing Airflow logs coverage and enhance it by adding logs wherever appropriately required.

### Related issues

Parent Issue: https://github.com/apache/airflow/issues/40975

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by KoviAnusha** [comment]
@potiuk @amoghrajesh @kaxil, can I work on this task?

**Comment by potiuk** [comment]
I guess that one is quite substantial and it requires quite an extensive knowledge of various components and airflow and logging approach. @KoviAnusha -> If that's one of your first tasks, then I would not encourage this one.

**Comment by KoviAnusha** [comment]
@potiuk, yes, this will be my first task. As suggested, I will take up other tasks in parallel improve my knowledge to take up such tasks.

**Comment by Dev-iL** [comment]
One thing I came across recently:

When we have code such as below in an operator, and unless one adds an explicit logging call, the task ends up being skipped silently with no indication of the reason (not even a "`AirflowSkipException` encountered - skipping"). I would expect a textual rendering of the "upstream" (custom) exception at the very least.

```python
raise AirflowSkipException from CustomExceptionWithParameter(param1)
```

**Comment by Dev-iL** [comment]
I noticed that when a task is configured to use a pool that doesn't exist - tasks get stuck indefinitely with no indication of why.

**Comment by potiuk** [comment]
> I noticed that when a task is configured to use a pool that doesn't exist - tasks get stuck indefinitely with no indication of why.

I don't think we even have place for logs for that - because essentially what happens is scheduler runs a join with `pool` table and this is how it selects tasks to run - so if you want to fix this one @Dev-iL , this is a new feature to add rather than logging - figure out a place somewhere where pool_id for tasks is checked against pool table.
